src/editor/utils/datainterface.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module core/editor/utils/datainterface
+ */
+
+/**
+ * Interface provides methods for setting and getting data to/from editor main root element of the model tree
+ * using {@link module:core/editor/editor~Editor#data data pipeline}.
+ *
+ * @mixin DataInterface
+ * @implements module:core/editor/utils/datainterface~Data
+ */
+const DataInterface = {
```

Comment:
So far we called all mixins "mixins". Like `ObservableMixin` which implements the `Observable` interface. So we didn't use the word `Interface`.

Also, a simple `Data` interface is kinda confusing. Editor, by implementing the `Data` interface becomes a data itself? The name needs to be clearer.

I'd propose something like:

* `EditorDataApi` interface
* `EditorDataApiMixin` mixin

Reply:
I'm fine with `EditorDataApiMixin`. However, note that these are in the `editor/utils` file and will be imported in the editor implementation class, so I believe it will be clear for everybody who imports it that, use it or see it there that it's about `DataApi` for the editor, not for something else. This is why `DataApiMixin` is fine for me as well.

Reply:
👍  for `DataApiMixin`. This will be consistent with our approach to not duplicate module's path in class names.